prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -754,24 +754,55 @@ class ParserSpec extends AnyFunSpec with Matchers {
   }
 
   it("should correctly parse durations with multiple units") {
-    {
-      val query = """foo{label="bar"}[1m30s] offset 2h15m"""
-      val windowSec = 60 + 30
-      val offsetSec = (2 * 60 * 60) + (15 * 60)
-      Parser.parseQuery(query) match {
+    case class Spec(query: String, windowSec: Int, offsetSec: Int)
+    val specs = Seq(
+      Spec(
+        """foo{label="bar"}[1m30s] offset 2h15m""",
+            60 + 30,
+            (2 * 60 * 60) + (15 * 60)
+      ),
+      Spec(
+        """foo{label="bar"}[3d2h25m10s] offset 2d12h15m30s""",
+            (3 * 24 * 60 * 60) + (2 * 60 * 60) + (25 * 60) + 10,
+            (2 * 24 * 60 * 60) + (12 * 60 * 60) + (15 * 60) + 30
+      ),
+      Spec(
+        """foo{label="bar"}[3d0h25m0s] offset 0d12h15m30s""",
+            (3 * 24 * 60 * 60) + (25 * 60),
+            (12 * 60 * 60) + (15 * 60) + 30
+      )
+    )
+    specs.foreach{ spec =>
+      Parser.parseQuery(spec.query) match {
         case RangeExpression(_, _, window, offset) =>
-          window.millis(0) shouldEqual 1000 * windowSec
-          offset.get.millis(0) shouldEqual 1000 * offsetSec
+          window.millis(0) shouldEqual 1000 * spec.windowSec
```

Comment:
looks good to me.
However, a minor thing I want to bring up.
The test for the original code would be
window.millis(0) / 1000  shouldEqual spec.windowSec.
There is no difference for the test because the milliseconds are divisible by 1000

Reply:
You're saying it would be more correct to use this, instead?
```
window.millis(0) / 1000 shouldEqual spec.windowSec
```

Reply:
Also in hindsight, it would be way clearer to use `Duration`s here-- I'll do that